### ISSUE TITLE
Validation epochs for RL models

### ISSUE THREAD

**Issue by djbyrne** [PyTorchLightning/pytorch-lightning-bolts#87]
## 🚀 Feature
Currently there are only train and test functionality. The models should also have a validation run per pseudo epoch

### Motivation

The metrics are currently based on the moving avg training reward, but that doesn't show the true performance of the agent, especially in the case of the e-greedy agents.

### Pitch
Every 1000 training epochs (which is essentially a step with the iterable dataset) a series of validation episodes should be run

### Additional context
Previously, I found some complications when trying to get the validation runs working with the iterabel dataset, however it seems that recent updates in PL should resolve these issues
